src/standalone/Soil/energy_hydrology.jl

```diff
@@ -23,8 +23,8 @@ $(DocStringExtensions.FIELDS)
 """
 Base.@kwdef struct EnergyHydrologyParameters{
     FT <: AbstractFloat,
-    F <: Union{<:AbstractFloat, ClimaCore.Fields.Field},
-    SF <: Union{<:AbstractFloat, ClimaCore.Fields.Field},
+    F <: Union{<:FT, ClimaCore.Fields.Field},
+    SF <: Union{<:FT, ClimaCore.Fields.Field},
```

Comment:
There's only one subtype of `FT` ... `FT` itself, so maybe you want to write `F <: Union{FT, ClimaCore.Fields.Field},`